neps/nep-0364.md

```diff
@@ -164,6 +173,8 @@ And a `rust-sdk` possible implementation could look like this:
 pub fn ed25519_verify(sig: &ed25519::Signature, msg: &[u8], pub_key: &ed25519::Public) -> bool;
 
 ```
+The current implementation is imported from the crate `ed25519-dalek`, version 1. It uses no feature flags
+except from the default one. This is the exact same setup used in the crate `near-crypto`.
```

Comment:
@abacabadabacaba Can you maybe help to identify how to state this requirement in a crate-agnostic way, so we don't get tied to the exact implementation?

Reply:
@frol It is difficult to describe the specific behavior concisely without referring to a specific implementation. [This blog post](https://hdevalence.ca/blog/2020-10-04-its-25519am) describes the various ways in which the existing Ed25519 implementations differ in practice. The behavior that we are using, which is shared by Go `crypto/ed25519`, Rust `ed25519-dalek` (using `verify` function with `legacy_compatibility` feature turned off) and several others, makes the following decisions:

* The encoding of the values $R$ and $s$ must be canonical, while the encoding of $A$ doesn't need to.
* The verification equation is $R=[s]B-[k]A$.
* No additional checks are performed. In particular, the points outside of the order-<span></span>$l$ subgroup are accepted, as are the points in the torsion subgroup.

While the above hopefully describes the behavior that we seek in an unambiguous manner, we may still want to point to a specific implementation to refer to in case of an ambiguity.

Reply:
@abacabadabacaba Thank you!

@blasrodri Please, include the proposed details into the NEP

Reply:
@frol updated it

Reply:
@blasrodri Probably Security Implications is not the most appropriate place for this, this should probably rather go just above this thread. Also, whenever the `ed25519-dalek` crate is mentioned, it should be specified that `verify` function should be used. This is because there are other verification functions in this crate (`verify_strict`, `verify_batch`) that behave differently.